File: triads.py
#!/usr/bin/env python3
import time
import sys
import random

from AppKit import NSSpeechSynthesizer

notes = [
    "A",
    "B",
    "C",
    "D",
    "E",
    "F",
    "G",
]

modifiers = [
    "", "sharp", "flat"
]

qualities = [
    "major", "minor", "diminished", "augmented"
]

interval = 5
if len(sys.argv) > 1:
    try: 
        interval = int(sys.argv[1])
    except ValueError:
        print("interval {} is not a number, using default.".format(sys.argv[1]))
        pass

ss = NSSpeechSynthesizer.alloc().init()

while True:
    note = random.choice(notes)
    modifier = random.choice(modifiers)
    quality = random.choice(qualities)
    triad = "[[char LTRL]] " + note + " [[char NORM]] " + modifier + " " + quality
    ss.startSpeakingString_(triad)
    time.sleep(interval)

File: triads.js
var notes = [
    "A",
    "B",
    "C",
    "D",
    "E",
    "F",
    "G",
];

var modifiers = [
    "", "♯", "♭"
];

var qualities = [
    "maj", "min", "dim", "aug"
];

var pickRandom = function(arr) {
    return arr[Math.floor(Math.random() * arr.length)];
}

var app = new Vue({ 
    el: '#app',
    data: {
        triads: []
    },
    methods: {
        generate: function(event) {
            this.triads = [];
            for (let i = 0; i < 10; i++) {
                this.triads.push({
                    note: pickRandom(notes),
                    modifier: pickRandom(modifiers),
                    quality: pickRandom(qualities)
                });
            }
        }  
    },
});
